contracts/StakingVault/RibbonStakingVault.sol

```diff
@@ -0,0 +1,637 @@
+// SPDX-License-Identifier: MIT
+pragma solidity =0.8.4;
+
+import {SafeMath} from "@openzeppelin/contracts/utils/math/SafeMath.sol";
+import {IERC20} from "@openzeppelin/contracts/token/ERC20/IERC20.sol";
+import {
+  SafeERC20
+} from "@openzeppelin/contracts/token/ERC20/utils/SafeERC20.sol";
+import {
+  ReentrancyGuardUpgradeable
+} from "@openzeppelin/contracts-upgradeable/security/ReentrancyGuardUpgradeable.sol";
+import {
+  OwnableUpgradeable
+} from "@openzeppelin/contracts-upgradeable/access/OwnableUpgradeable.sol";
+import {
+  RibbonStakingVaultStorage
+} from "../storage/RibbonStakingVaultStorage.sol";
+import {Vault} from "../libraries/Vault.sol";
+import {VaultLifecycle} from "../libraries/VaultLifecycle.sol";
+import {ShareMath} from "../libraries/ShareMath.sol";
+import {IWETH} from "../interfaces/IWETH.sol";
+import {ISRBN} from "../interfaces/ISRBN.sol";
+
+/**
+ * UPGRADEABILITY: Since we use the upgradeable proxy pattern, we must observe
+ * the inheritance chain closely.
+ * Any changes/appends in storage variable needs to happen in RibbonThetaVaultStorage.
+ * RibbonThetaVault should not inherit from any other contract aside from RibbonVault, RibbonThetaVaultStorage
+ */
+contract RibbonStakingVault is
+  ReentrancyGuardUpgradeable,
+  OwnableUpgradeable,
+  RibbonStakingVaultStorage
```

Comment:
Since we are only upgrading the storage in `RibbonStakingVaultStorage`, we need to move all the state variables into `RibbonStakingVaultStorage`.

The reason is that if we added new storage variables into `RibbonStakingVaultStorage`, it would break the storage layout.

How it currently is based on the inheritance chain:

```
ReentrancyGuardUpgradeable -> OwnableUpgradeable -> RibbonStakingVaultStorage (upgrading this breaks the subsequent storage variables) -> RibbonStakingVault
```